central/notifications/service/service_test.go

```diff
@@ -0,0 +1,197 @@
+package service
+
+import (
+	"context"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	dsMocks "github.com/stackrox/rox/central/notifications/datastore/mocks"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/protoconv"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/mock/gomock"
+)
+
+var (
+	errFake          = errors.New("fake error")
+	fakeID           = "0925514f-3a33-5931-b431-756406e1a008"
+	fakeNotification = &storage.Notification{
+		Id:             fakeID,
+		Level:          storage.NotificationLevel_NOTIFICATION_LEVEL_DANGER,
+		Message:        "message",
+		Type:           storage.NotificationType_NOTIFICATION_TYPE_GENERIC,
+		Hint:           "hint",
+		Domain:         "domain",
+		NumOccurrences: 1,
+	}
+	maxPagination = &v1.Pagination{
+		Limit: math.MaxInt32,
+	}
+)
+
+func TestNotificationService(t *testing.T) {
+	t.Parallel()
+	suite.Run(t, new(countNotificationsTestSuite))
+	suite.Run(t, new(getNotificationTestSuite))
+	suite.Run(t, new(listNotificationsTestSuite))
+}
+
+type baseTestSuite struct {
+	suite.Suite
+	mockCtrl *gomock.Controller
+
+	ctx           context.Context
+	datastoreMock *dsMocks.MockDataStore
+	service       Service
+}
+
+func (s *baseTestSuite) SetupTest() {
+	s.mockCtrl = gomock.NewController(s.T())
+	s.ctx = context.Background()
+	s.datastoreMock = dsMocks.NewMockDataStore(s.mockCtrl)
+	s.service = newService(s.datastoreMock)
+}
+
+func (s *baseTestSuite) TearDownTest() {
+	s.mockCtrl.Finish()
+}
```

Comment:
Same comment as in other PRs w.r.t [gomock.Controller.Finish](https://pkg.go.dev/github.com/golang/mock/gomock#Controller.Finish) not being required.